src/lua/httpc.lua

```diff
@@ -329,6 +335,122 @@ local function encode_url_params(params, http_method)
     return res
 end
 
+local function check_limit(self, limit)
+    if self._internal.rbuf:size() >= limit then
+        return limit
+    end
+    return nil
+end
+
+local function check_delimiter(self, limit, eols)
+    if limit == 0 then
+        return 0
+    end
+    local rbuf = self._internal.rbuf
+    if rbuf:size() == 0 then
+        return nil
+    end
+
+    local shortest
+    for _, eol in ipairs(eols) do
+        local data = ffi.C.memmem(rbuf.rpos, rbuf:size(), eol, #eol)
```

Comment:
memmem(3) is not specified in POSIX.1. Is it OK for us?

Reply:
The code is copy-pasted from socket.lua. So I think it is OK here.